frontend/routes/viewerGui/components/groups/components/groupDetails/groupDetails.component.tsx

```diff
@@ -108,82 +112,93 @@ export class GroupDetails extends React.PureComponent<IProps, IState> {
 
 	public renderPreview = renderWhenTrue(() => (
 		<PreviewDetails
-			key={this.groupData._id}
-			{...this.groupData}
-			roleColor={this.groupData.color}
-			created={this.groupData.createdAt}
+			key={this.editingGroup._id}
+			{...this.editingGroup}
+			roleColor={this.editingGroup.color}
+			created={this.editingGroup.createdAt}
 			editable={this.props.canUpdate}
 			onNameChange={this.handleFieldChange}
 			renderCollapsable={this.renderGroupForm}
-			renderNotCollapsable={() => this.renderRulesField(this.groupData.type === GROUPS_TYPES.SMART)}
+			renderNotCollapsable={() => this.renderRulesField(this.editingGroup.type === GROUPS_TYPES.SMART)}
 			panelName={GROUP_PANEL_NAME}
-			isSmartGroup={this.groupData.type === GROUPS_TYPES.SMART}
-			StatusIconComponent={GROUP_TYPES_ICONS[this.groupData.type]}
+			isSmartGroup={this.editingGroup.type === GROUPS_TYPES.SMART}
+			StatusIconComponent={GROUP_TYPES_ICONS[this.editingGroup.type]}
 			scrolled={this.state.scrolled}
 			isNew={this.isNewGroup}
 		/>
 	));
 
+	public areSharedIdsChanged = (selectedNodes = [], groupObjects = []) => {
+		const toFullIdsDict = (dict, val) => {
+			val.shared_ids.forEach((e) => dict[val.account + '.' + val.model + '.' + e] = true);
+			return dict;
+		};
+
+		selectedNodes = selectedNodes.reduce(toFullIdsDict, {});
```

Comment:
My worry is when the number of objects get huge (and it can be if someone has a massive federation and decided to select all the objects somehow...), this will grind to a halt.

I think we need some early rejects, like if the lengths doesn't compare (both the account/model level and the number of shared Ids within each account/model entry)

And I think we can assume the order as well (unity keep these as a list and just spit it out anyway so the order can be assumed) - so I don't think we need the reduction. It's much better to act quicker but potentially false positives than to grind to a halt every time.


This is definitely neater than what I'm proposing, but speed matters here i think!

Reply:
Unfortunately we cant assume order, as it depends on the order which the meshes were selected.

Reply:
Yea, what I mean is if the order isn't right even if the same set of mesh we just return and say something has changed, as that would only be caused by user has done some interaction with the viewer.

Reply:
it's a false positive, but it's better than going through with the whole process... I think.